app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb

```diff
@@ -197,4 +199,26 @@ def sshkeys
       persister.auth_key_pairs.build(:name => tenant_key[:name])
     end
   end
+
+  def systemtypes
+    collector.system_pool.each do |v|
+      persister.flavor.build(
+        :types   => ::ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::SystemType,
+        :ems_ref => v['type']
+      )
+    end
+  end
+
+  def storagetypes
+    region = collector.region
+    collector.storage_types[region].each do |v|
```

Comment:
If we're using `collector.region` and passing it back to `collector.storage_types[region]` could we just do `connection.get_storage_types[region]` unless you want to be able to get storage types from other regions elsewhere

Reply:
again, I initially thought of doing it there, but decided as I wanted to do no parsing in the collector functionality. But it can be put there. Either way this changes when the backend finalizes/corrects their web api. I can move it in the collector code though.